components/Footer.tsx

```diff
@@ -198,6 +200,7 @@ type ThemeTypes = "dark" | "light" | "system";
 const Footer = () => {
   const [localStorageTheme, setLocalStorageTheme] = useState<ThemeTypes>();
   const theme = useTheme();
+  const preferredTheme = usePreferredTheme() || localStorageTheme;
```

Comment:
tbh, we should just remove the useState for localStorageTheme, I wrote this hook a bit ago so sorry if im not remembering correctly, but `usePreferredTheme` listens for storage events which renders the state kinda useless.

Reply:
is it what you needed? At Front I'm a vue.js developer, react syntax sounds a little different...

Reply:
bingo! exactly what i mean. looks good! thank you!